packages/babel-traverse/src/path/index.ts

```diff
@@ -43,7 +43,7 @@ class NodePath<T extends t.Node = t.Node> {
     this.scope = null;
   }
 
-  declare parent: t.ParentMaps[T["type"]];
+  declare parent: t.Node;
```

Comment:
Does parent path type narrowing still work in this PR?

Reply:
Yes, because it's properly defined further down in the `NodePath` interface.